frontend/src/registration/components/DomainInputPage/DomainInputPage.tsx

```diff
@@ -1,51 +1,58 @@
 import { ROUTES } from 'global/constants';
 import { RootState } from 'global/state';
 import * as React from 'react';
-import {
-  Button,
-  Form,
-  FormControl,
-  FormGroup,
-  FormLabel,
-  InputGroup
-} from 'react-bootstrap';
 import { connect } from 'react-redux';
 import { WrappedMessage } from 'utils/intl';
-import { InstitutionalAccountHero } from 'ui/components';
-import { submitRegistration } from '../../actions';
+import { DomainInput, InstitutionalAccountHero } from 'ui/components';
+import { submitRegistration, updateDomainInfoState } from '../../actions';
 import { RegistrationPage } from '../RegistrationPage';
+import { getRegistrationData } from '../../selectors';
 import messages from './displayMessages';
+import './styles.scss';
 
 interface ActionProps {
   submitRegistration: Function;
+  updateDomainInfoState: Function;
 }
 
 interface Props extends ActionProps {}
 
-interface State {
-  domainName: string;
+interface StateProps {
+  domain: string;
+  domainIsExternal: boolean;
+  domainError: string;
+  loading: boolean;
 }
 
-@connect<{}, ActionProps, {}, Props, RootState>((state: RootState) => ({}), {
-  submitRegistration
-})
-export class DomainInputPage extends React.PureComponent<Props, State> {
-  public constructor(props: Props, state: State) {
+interface Props extends StateProps, ActionProps {}
+
+@connect<{}, ActionProps, {}, Props, RootState>(
+  (state: RootState) => ({
+    domain: getRegistrationData(state, 'domain'),
+    domainIsExternal: getRegistrationData(state, 'domainIsExternal'),
+    domainError: getRegistrationData(state, 'domainError'),
+    loading: getRegistrationData(state, 'loading')
+  }),
+  {
+    submitRegistration,
+    updateDomainInfoState
+  }
+)
+export class DomainInputPage extends React.PureComponent<Props> {
+  public constructor(props: Props) {
     super(props);
-    this.state = {
-      domainName: ''
-    };
   }
 
-  private domainNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({
-      domainName: event.target.value || ''
+  private handleDomainChange = (newDomain: string) => {
+    this.props.updateDomainInfoState({
+        domain: newDomain,
+        domainError: ''
     });
   };
 
-  private submitForm = () => {
+  private submitDomain = () => {
     this.props.submitRegistration(
-      { domain: this.state.domainName },
+      { domain: this.props.domain },
       ROUTES.Registration.INSTANCE
     );
```

Comment:
Are we planning to do some domain validation on the frontend?
Something like `^[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$` should work.